http_queue_service.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+type httpClient interface {
+	Do(req *http.Request) (resp *http.Response, err error)
+}
+
+type httpQueue struct {
+	httpClient httpClient
+	endpoint   string
+}
+
+func newHttpQueueService(httpClient httpClient, endpoint string) httpQueue {
+	return httpQueue{
+		httpClient: httpClient,
+		endpoint:   endpoint,
+	}
+}
+
+func (q httpQueue) sendMessage(id string, conceptType string, tid string, payload []byte) error {
+
+	url, err := url.Parse(strings.TrimRight(q.endpoint, "/") + "/" + id)
```

Comment:
Yup.  url.Parse should be able to support it but it's safer to avoid it in the first place.